crates/matrix-sdk-crypto/src/session_manager/group_sessions.rs

```diff
@@ -650,6 +651,88 @@ impl GroupSessionManager {
         Ok(())
     }
 
+    fn log_room_key_sharing_result(
+        session: &OutboundGroupSession,
+        requests: &[Arc<ToDeviceRequest>],
+    ) {
+        #[cfg(feature = "message-ids")]
```

Comment:
nit: There's a lot of `cfg(feature)` statements in this function which make it a bit hard to read, would it make sense to "hoist" the cfg guards by having two different implementations of `log_room_key_sharing_result`, at the risk of slightly duplicating the loop's body?

Reply:
I pondered this as well, but decided that it would lead to too much duplication. The only difference here is that the map is different and thus also the insertion into the map slightly different.

Since the logging is already put into its separate method I think we can live with this.